### ISSUE TITLE
New windows are very tiny on high-DPI monitors on Windows

### ISSUE THREAD

**Issue by hrydgard** [openscad/openscad#2603]
When opening files or just starting the program on a high-dpi monitor, windows are created tiny and icons are unnecessarily small. I'm running at 225% scale in Windows settings. The windows can be manually resized, but this doesn't fix the icon problem and is simply a bit annoying. 

Here's what the windows look like on my machine. Note that while they look big here on Github because it scales pictures to some kind of webpage DPI, when the screenshots were taken on my display they were very tiny.

![intro window](https://user-images.githubusercontent.com/130929/48970989-7d7a9d80-f014-11e8-8ce5-9975f5a2168c.JPG)

![high dpi small window](https://user-images.githubusercontent.com/130929/48970990-7f446100-f014-11e8-8578-7cfd38da4974.JPG)

![small config window](https://user-images.githubusercontent.com/130929/48970993-823f5180-f014-11e8-98b5-962a571a0050.JPG)

**Comment by t-paul** [comment]
That is something which should be handled by the underlying UI framework. Can you please try how the [development snapshots](http://www.openscad.org/downloads.html#snapshots) behave? Those should come with a newer version of that framework which is supposed to handle per monitor DPI scaling.

**Comment by hrydgard** [comment]
The development snapshot behaves the same.

**Comment by MichaelAtOz** [comment]
Which  Windows version?
I suspect W7? W7 doesn't handle high DPI properly, apparently W10 does, but I just live with small icons/fonts.

**Comment by hrydgard** [comment]
Nah, this is Windows 10.

Except for the initial window sizes and small icons, it all seems to work fine, so this is a mild annoyance. Does smell like hardcoded window sizes in pixels somewhere though.

**Comment by t-paul** [comment]
Nope, it's not hardcoded (well, the UI files have dimensions, but scaling DPI scaling should still work). It does work as advertised on Linux using the documented environment variable. If that is not set, I'm getting the same effect on my Dell XPS 13 @ 3840x2160.

Docs linked above say:
> Qt applications by default are Per-Monitor DPI Aware on Windows 8.1 or System-DPI Aware on older versions of Windows. As of Qt 5.4, the level can be specified by passing a parameter to the platform plugin (see Using qt.conf).

I guess we need to check if there's some unwanted config shipped somehow.

**Comment by MichaelAtOz** [comment]
That page also  says 
_Applications mostly work with device independent pixels. **Notable exceptions are OpenGL** and code that works with raster graphics._
?

I can't recall what it did with W7 @ 150% DPI (to make fonts more readable), but I didn't like it. I'll try again.

**Comment by ksuszka** [comment]
I installed a fresh installation of Windows 10 and OpenSCAD 2021.01, set system scalling to 225% and stumbled on the same issue. What solved it for me was setting QT_AUTO_SCREEN_SCALE_FACTOR environment variable to 1 before running OpenSCAD. This variable is described here: https://doc.qt.io/qt-5/highdpi.html#high-dpi-support-in-qt.

**Comment by erjoalgo** [comment]
QT_AUTO_SCREEN_SCALE_FACTOR didn't seem to work for me, but QT_SCALE_FACTOR=3 did

**Comment by MichaelAtOz** [comment]
I just looked at the above link, now it says:

The QT_ENABLE_HIGHDPI_SCALING environment variable, introduced in Qt 5.14, enables automatic scaling based on the pixel density of the monitor. Replaces QT_AUTO_SCREEN_SCALE_FACTOR.
